classes/controller/ProductListingFrontController.php

```diff
@@ -497,24 +500,48 @@ protected function getAjaxProductSearchVariables()
     /**
      * Cleans the products array with only whitelisted properties.
      *
-     * @return array
+     * @param array[] $products
+     *
+     * @return array[] Filtered product list
      */
     protected function prepareProductArrayForAjaxReturn(array $products)
     {
-        $allowed_properties = array('id_product', 'price', 'reference', 'active', 'description_short', 'link',
-            'link_rewrite', 'name', 'manufacturer_name', 'position', 'cover', 'url', 'canonical_url', 'add_to_cart_url',
-            'has_discount', 'discount_type', 'discount_percentage', 'discount_percentage_absolute', 'discount_amount',
-            'price_amount', 'regular_price_amount', 'regular_price', 'discount_to_display', 'labels', 'main_variants',
-            'unit_price', 'tax_name', 'rate'
-        );
-        foreach ($products as $product_key => $product) {
-            foreach ($product as $product_property => $data) {
-                if (!in_array($product_property, $allowed_properties)) {
-                    unset($products[$product_key][$product_property]);
-                }
-            }
-        }
-        return $products;
+        $filter = (new CollectionFilter())
+            ->queue([
+                (new HashMapWhitelistFilter())
```

Comment:
Shouldnt be ProductFilter for consistency?

Reply:
If not, how about create a specific and overiddable class/object/service instead? I feel we should not write configuration in controller :(

Reply:
Yeah you're right, I'll make a specific class for this.